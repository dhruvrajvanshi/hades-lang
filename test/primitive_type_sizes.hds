import submodule.test_utils as utils;

fn print_size[T](name: *Byte): Void {
    utils.print_byte_string(name);
    utils.print_byte_string(b": ");
    utils.print_size(size_of[T]);
}

fn main(): Void {
    print_size[u8](b"u8");
    print_size[i8](b"i8");
    print_size[u16](b"u16");
    print_size[i16](b"i16");
    print_size[u32](b"u32");
    print_size[i32](b"i32");
    print_size[u64](b"u64");
    print_size[i64](b"i64");
    print_size[f32](b"f32");
    print_size[f64](b"f64");
}
