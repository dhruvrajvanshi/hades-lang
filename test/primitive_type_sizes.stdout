u8: 1
i8: 1
u16: 2
i16: 2
u32: 4
i32: 4
u64: 8
i64: 8
f32: 4
f64: 8
